### ISSUE TITLE
"commitsSort": false does not work as documented

### ISSUE THREAD

**Issue by henrikvilhelmberglund** [absolute-version/commit-and-tag-version#48]
I'm probably stupid but I want to have  "commitsSort": false for sorting commits by date but I'm not sure how to do it with my global npm install of commit-and-tag-version. How would I accomplish that?

Also, would "commitsSort": false generate a new changelog with all commits sorted by date or would it be a retroactive thing from when I changed the option?

**Comment by henrikvilhelmberglund** [comment]
Not really, I just saw this in the readme:
> This has the benefit of allowing you to use commit-and-tag-version on any repo/package without adding a dev dependency to each one.

and thought it would be more convenient and didn't really think about configuring stuff.

Anyway I removed the global install and installed it as a dev dependency instead but it didn't really change things, commits are still sorted alphabetically with this in the .versionrc:
```json
{
  "commit-and-tag-version": {
    "writerOpts": {
      "commitsSort": false
    }
  }
}
``` 

I'm guessing I need something else as well?

**Comment by TimothyJones** [comment]
Oh! Haha! I didn't read the readme too closely when I forked it, so I missed that recommendation.

I think global installs have kind of fallen out of fashion, as they generally cause dependency issues further down the track-  perhaps we should remove that recommendation. What do you think?

As to your question, I'll take a look at the option and get back to you.

Note that the changelog is not regenerated from scratch, so changing the options will only affect future releases.

**Comment by TimothyJones** [comment]
I can confirm that `"commitsSort": false` doesn't work as documented. The code looks right to me, and it seems to be poked through to conventional-commits-writer as expected. It needs some time with a debugger - I'll spend another 30 mins on it now, and if that doesn't find the problem, I'll set aside some more time over the next couple of weeks.

**Comment by TimothyJones** [comment]
Update:  it seems to work when installed as a `devDependency`, but not when *not* installed and run with `npx commit-and-tag-version`. I would guess that it's not picking up the package.json when run only with `npx` - but I know that it is reading from package.json when run only with `npx`, as that's how I investigated #47.

As a workaround, you can install it as a `devDependency`. This still requires further investigation to find out why it doesn't work when not installed directly.

**Comment by TimothyJones** [comment]
Aha! Got it - I think `npx` caches the versions it installs - if you do `npx commit-and-tag-version@10.1.0` it should install the latest version which passes the writer options down.

I'll close this for now, please reopen if this doesn't fix the problem for you.

**Comment by henrikvilhelmberglund** [comment]
It didn't solve the problem (still had alphabetical commits) but I had the writerOpts in the wrong place (.versionrc instead of .package-json).

I had no idea you could even set config options in the package.json (seems a bit janky honestly) but now it works and I'm happy. ☺️ 

I did kind of make a mess of the changelog/tags though, is there a good way to fix these?

**Comment by TimothyJones** [comment]
It will work in .versionrc- just drop the “commit-and-tag-version” and use:

```
{
    "writerOpts": {
      "commitsSort": false
    }
}
```

directly. The options are the same in package.json or in the versionrc- it’s just a matter of preference.

What kind of mess did you make in the changelog? Probably the easiest strategy is to wind it back and force push (you may need to also delete local and remote tags).

If you need to trial different settings again, you can use `—dry-run` to try out different settings on the changelog without stamping on the history

**Comment by henrikvilhelmberglund** [comment]
Oh yeah, I tried the .versionrc method when I had the global install but not after changing to a local install, I changed to that now and it worked fine.

I also fixed the changelog, I removed the tags on github, changed the version in the package and tried to rerun it but it failed because I still had git tags locally, after removing those it's now fine again.

Thanks a lot for the help!
